mautic/mautic
# adding to salesforce campaign members from Mautic campaign

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR creates a new fields in the push to integration action in a campaign to be able to add campaign contact to an integration campaign.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. 
2. 

#### Steps to test this PR:
1.Apply this PR, make sure you have salesforce plugin enabled and fields matched 
2. create a new campaign with push to integration action.
3. Select salesforce and see new dropdown fields (Campaign, Campaign contact status)
4. Select a campaign and a campaign contact status.
5. Trigger the campaign and contacts from Mautic should be pushed to Salesforce and put in the Salesforce campaign selected as campaign members
6. Test the following scenarios: 
a. If a campaign member doesn't exists Mautic should create the campaign member, if it exists is should update it with it's new status  if status is sent.
 b. A campaign member that was previously pushed by Mautic was deleted from a campaign in SF, the campaign member should be created again if pushed by a different campaign or same campaign in Mautic.
c. if there is a contact in Mautic and this contact's email address exists multiple times in SF (as a Lead, converted Lead or Contact) all matching SF leads or contacts should be created as campaign members in SF
